### PR TITLE
Add python package to use schemas

### PR DESCRIPTION
**Pull request by mythmon**
This is just a bare-bones schema checker for now, but it should make a Python package that can be used in other projects, and it integrates with the build system.

### CODE REVIEW

**Comment by mythmon** [comment]
@jaredlockhart I think that this should be ready for some initial testing. The uses are pertty basic, with just a `check_schema` method. I've updated the README to explain how to use the new package, and you should be able to install it from the artifacts of this PR:

```shell
$ poetry add https://90-267090370-gh.circle-artifacts.com/0/artifacts/python/mozilla_nimbus_shared-0.0.1-py3-none-any.whl
```

Can you try and install the package and see if it works on your end? Does the API presented make sense? Anything else you'd like to see? Feel free to review the implementation and generation code, but I only need reviews on the public interface half of this.

File: README.md [hunk 1]
```diff
@@ -87,9 +87,25 @@ typeGuards.normandy_assertAddonRollbackArguments(userInput);
...
 ### Python
 
-Native support coming soon. For now, consider using the NPM package and loading the schemas directly
-from the filesystem. After installing this package via `npm`, the schemas will be available at
-`node_modules/@mozilla/rapid-experiments-shared/schemas/<group>/<type>.json`.
+Basic native support is available via the PyPI package `mozilla-nimbus-shared`. The only function is
+exposed is one to validate an object against a named schema.
+
+```python
+from mozilla_nimbus_shared import check_schema
```
Comment (pdehaan) [inline README.md:94]: Sorry, I don't Python very well. Why is it `mozilla-nimbus-shared` (hyphens) above, but `mozilla_nimbus_shared` (underscores) here?

File: Makefile [hunk 2]
```diff
@@ -1,58 +1,72 @@
...
+	@touch $(PYTHON_INSTALL_STAMP)
+
 $(TSC_STAMP): $(GENERATED_TS) $(TS_SRC) $(TYPES) $(NPM_INSTALL_STAMP) tsconfig.json
-	tsc
 	@mkdir -p $(@D)
+	tsc
 	@touch $@
 
-$(SCHEMAS): $(TYPES) $(NPM_INSTALL_STAMP) bin/build-schemas.ts
+$(SCHEMAS) &: $(TYPES) $(NPM_INSTALL_STAMP) bin/build-schemas.ts
```
Comment (pdehaan) [inline Makefile:69]: Seeing these warnings in CircleCI logs (here and on L84 below):

```
Makefile:85: warning: overriding recipe for target '&'
Makefile:70: warning: ignoring old recipe for target '&'
eslint .
```

File: README.md [hunk 3]
```diff
@@ -87,9 +87,25 @@ typeGuards.normandy_assertAddonRollbackArguments(userInput);
...
+
+# Returns True
+check_schema("normandy/ConsoleLogArguments", {"message": "hello, world!"})
+```
+
+### Other
+
+For other applications, consider using the NPM package and loading the schemas directly from the
+filesystem. After installing this package via `npm`, the schemas will be available at
+`node_modules/@mozilla/rapid-experiments-shared/schemas/<group>/<type>.json`. Please leave an issue
```
Comment (pdehaan) [inline README.md:107]: Nit: this needs a tweakin' from "rapid-experiments-shared" to "nimbus-shared", pre recent repo renamin'.

File: README.md (hunk 1 above)
Comment (mythmon) [reply README.md:94]: Package names are encouraged to by kebab-case, but module names must be valid identifiers. This is a pretty common pattern in Python-land.

**Comment by mythmon** [comment]
> a way to get a list of all schemas, maybe with a description?

Yes, definitely. This will be one of next improvements I want to make to both the JS and Python versions.

> when the validation fails it tells you exactly what's missing (which is AWESOME) but it also dumps out the entire schema, which is way too much noise to find what the actual error is, is there a way to not dump out the whole schema?

That's the default stringification of the error. If you catch the error you can get the individual bits:

```python
try:
    res = checkSchema("normandy/AddonRolloutArguments", {})
except ValidationError as e:
    print(e.message)
    # 'slug' is a required property
```

There are a couple problems here though, one is that it's only showing the first error. I'll need to do something like #27 again here. The second is that `ValidationError` isn't easy to get at.

I've filed these as #33 and #34

File: Makefile (hunk 2 above)
Comment (mythmon) [reply Makefile:69]: First off, thanks for looking through those logs. The `&:` syntax is an extension of GNU Make that allows you to say that multiple targets are all built together by the same command (as opposed to running the command once per target). I guess CI isn't using GNU Make like I am locally. That means that CI is probably doing some extra work rebuilding things, but it isn't a problem.

It shouldn't be harmful, but definitely something I'd like to fix. Lets do that in another PR though.

**Pull request merged**